### PR TITLE
Added entitlementSpec and configSpec to state.json

### PR DESCRIPTION
**Pull request by salahalsaleh1993**
What I Did
------------
Query entitlement spec and config spec to be included in state.json to be used for the Config tab instead of displaying the state.json as raw data, and to be used in the to-be-added Entitlements tab in Ship Cluster 

How I Did it
------------

How to verify it
------------

Description for the Changelog
------------

Picture of a Ship (not required but encouraged)
------------

### CODE REVIEW

File: pkg/specs/replicatedapp/graphql.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ const ShipRelease = `
     channelIcon
     semver
     releaseNotes
-    spec
+		spec
+		configSpec
```
Comment (laverya) [inline pkg/specs/replicatedapp/graphql.go]: indentation is broken here
I think you might have used tabs?

File: pkg/specs/replicatedapp/graphql.go [hunk 2]
```diff
@@ -43,7 +44,8 @@ const ShipRelease = `
         size
         data
       }
-    }
+		}
+		entitlementSpec
```
Comment (laverya) [inline pkg/specs/replicatedapp/graphql.go]: indentation is broken here too

File: contracts/replicatedapp/get_slug_release_test.go [hunk 3]
```diff
@@ -77,8 +77,53 @@ func Test_GetSlugRelease(t *testing.T) {
...
 						"images": []string{},
 						"githubContents": []string{},
+						"configSpec": dsl.Like(dsl.String("v1:\n  - name: test_options\n    title: Test Options\n    description: testing testing 123\n    items:\n      - name: test_option\n        title: Test Option\n        default: abc123_test-option-value\n        type: text\n")),
+						"entitlementSpec": dsl.Like(dsl.String("---\n- name: Has Contract\n  key: has_contract\n  type: boolean\n  default: false\n- name: Is SAML Allowed\n  key: is_saml_allowed\n  type: boolean\n  default: false\n- name: Has Private Support\n  key: is_support_ticket_allowed\n  type: boolean\n  default: false\n")),
 						"entitlements": map[string]interface{}{
-							"values": []string{},
+							"values": [3]map[string]interface{}{
```
Comment (laverya) [inline contracts/replicatedapp/get_slug_release_test.go]: Make sure that the provider state has this

Comment (salahalsaleh1993) [reply contracts/replicatedapp/get_slug_release_test.go]: i updated the fixtures in vandoor and tested the new pact json file there

File: pkg/specs/replicatedapp/graphql.go (hunk 2 above)
Comment (salahalsaleh1993) [reply pkg/specs/replicatedapp/graphql.go]: this is just on Github, dunno why

File: contracts/replicatedapp/get_slug_release_test.go (hunk 3 above)
Comment (laverya) [reply contracts/replicatedapp/get_slug_release_test.go]: Ok
Well let's wait for that API change to make it to staging then

File: integration/init_app/integration_test.go [hunk 4]
```diff
@@ -155,6 +155,8 @@ var _ = Describe("ship init replicated.app/...", func() {
 
 					ignoreEntitlementSig := map[string][]string{
 						".ship/state.json": {
+							"v1.upstreamContents.appRelease.configSpec",
+							"v1.upstreamContents.appRelease.entitlementSpec",
```
Comment (laverya) [inline integration/init_app/integration_test.go]: once again, the indentation here is wrong (and in `update`)

Comment (laverya) [reply integration/init_app/integration_test.go]: (and this was my view wrapping things - whoops)

**Pull request merged**